contracts/RicoToken.sol

```diff
@@ -2,19 +2,84 @@ pragma solidity ^0.5.0;
 
 import "./zeppelin/token/ERC777/ERC777.sol";
 
+interface ReversibleICO {
+    function getLockedTokenAmount(address) external returns (uint256);
+}
+
 contract RicoToken is ERC777 {
+
+    ReversibleICO public rICO;
+    address public manager;
+    bool public freezed;
+
     constructor(
         uint256 initialSupply,
-        address[] memory defaultOperators
+        address[] memory _defaultOperators
     )
-        ERC777("RicoToken", "RICO", defaultOperators)
+        ERC777("LYXeToken", "LYXe", _defaultOperators)
         public
     {
         _mint(msg.sender, msg.sender, initialSupply, "", "");
+        manager = msg.sender;
+        freezed = true;
+    }
+    function setup(address _rICO, address _newManager) public {
+        require(msg.sender == manager, "only manager");
+        rICO = ReversibleICO(_rICO);
+        manager = _newManager;
+        freezed = false;
```

Comment:
i guess we should call this `frozen`

Reply:
Yes, there are a few others as well that I'll go over and update once i'm done with the rICO